### ISSUE TITLE
Ignore comments and trim white space for the env parser

### ISSUE THREAD

**Issue by Fly1nP4nda** [kellyjonbrazil/jc#372]
Can the env parser be extended to also handle env files that may have comments (ignore the comments) and trim white spaces that may be between the key/value pairs.

```
MySql_Port = 3306
MySql_Hostname = localhost
```

That above produces this and will carry the leading/trailing whitespaces into the key/values

```
cat test.env | jc --env | jq
[
  {
    "name": "MySql_Port ",
    "value": " 3306"
  },
  {
    "name": "MySql_Hostname ",
    "value": " localhost"
  }
]
```

**Comment by kellyjonbrazil** [comment]
Hi - thanks for your question!

this is possible, but the issue I see is that is no longer a valid env file. Bash doesn’t allow spaces between the equal sign and I’m not sure if comments can be supported in all cases. I’ll have to think about comments more. 

For the type of file you are describing, the —kv (Key/Value) parser might work better. Could you see if that works for your use case?

**Comment by Fly1nP4nda** [comment]
That is a fair point and thanks for the heads up. This might be a change we'll need to make to our env files for our application(s) for consistency.

A little background: 
We currently have env files - think of them as template or base env's with non-sensitive information that we would like to parse into the environment block of an AWS ECS task definition. The env parser works great in this sense because it parses out into the name/value json object.

I did try --kv at first but I found that it was lower casing all the keys (an upgraded version of jc did fix that however)

In the meantime I did find a temporary work around

```
grep -v ^\# envwithcomments.env | jc --env
```
